client/dashboard/sites/staging-site-delete-banner/index.tsx

```diff
@@ -0,0 +1,88 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
+import {
+	__experimentalText as Text,
+	__experimentalHeading as Heading,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	Card,
+	CardBody,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect } from 'react';
+import { siteByIdQuery } from '../../app/queries/site';
+import { stagingSiteDeleteStatusQuery } from '../../app/queries/site-staging-sites';
+import PageLayout from '../../components/page-layout';
+import deleteStagingSiteIllustration from './delete-staging-site-illustration.svg';
+import type { Site } from '../../data/types';
+
+export default function StagingSiteDeleteBanner( { site }: { site: Site } ) {
+	const router = useRouter();
+	const queryClient = useQueryClient();
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const productionSiteId = site.options?.wpcom_production_blog_id;
+
+	// Poll for staging site deletion status
+	const { data: stagingSiteData } = useQuery( {
+		...stagingSiteDeleteStatusQuery( productionSiteId ?? 0 ),
+		enabled: !! productionSiteId,
+	} );
+
+	// Fetch production site data for redirect
+	const { data: productionSite } = useQuery( {
+		...siteByIdQuery( productionSiteId ?? 0 ),
+		enabled: !! productionSiteId,
+	} );
+
+	// Redirect to the production site when the staging site is deleted
+	useEffect( () => {
+		if (
+			stagingSiteData &&
+			Array.isArray( stagingSiteData ) &&
+			stagingSiteData.length === 0 &&
+			productionSite?.slug
+		) {
+			// Clear the mutation state to prevent banner from showing after redirect
+			queryClient.getMutationCache().clear();
+
+			// Also clear the staging site delete status query
+			queryClient.removeQueries( {
+				queryKey: [ 'staging-site-delete-status', productionSiteId ],
+			} );
```

Comment:
I think clearing the entire mutation cache can be a bit dangerous. Using query invalidation instead should work.

```suggestion
			// Invalidate staging site deletion mutation and related queries
			queryClient.invalidateQueries( {
				queryKey: [ STAGING_SITE_DELETE_MUTATION_KEY, site.ID ],
			} );

			// Invalidate the status polling query
			queryClient.invalidateQueries( {
				queryKey: [ 'staging-site-delete-status', productionSiteId ],
			} );
```

Reply:
This suggestion was my original approach, but I wasn't able to remove the mutation from the cache using `invalidateQueries` unfortunately. This caused the banner to still be shown after the redirection (at around 0:33 in the video):

https://github.com/user-attachments/assets/b6d1bc57-8c8f-4b08-928c-cf21d348194c

I didn't find a function in the Tanstack Query docs that lets me remove/invalidate just one mutation in the cache.

Reply:
Ah, I found the solution to removing a specific mutation in [this thread](https://github.com/TanStack/query/discussions/2768#discussioncomment-1566635). This works great, I updated it in e37313d39fdc174d238f14f881b78151f5993ecb